doc/audit/SONiC_Audit_Enhancement.md

```diff
@@ -0,0 +1,149 @@
+  - `config audit enable <name>.rules`
+  - `config audit disable <name>.rules`
+
+### 3.2 Rules Review
+###### Table 2: Rules Review
+| Rule name | Details |
+|--------------------------|--------------------------|
+| Process audit            | `-a never,exit -F path=/usr/bin/runc -F perm=x`<br>`-a never,exit -F path=/usr/bin/python* -F perm=x`<br>`-a never,exit -F path=/usr/bin/docker -F perm=x` |
+| File deletion            | `-a exit,always -F arch=b64 -S unlink -S unlinkat -F key=file_deletion`<br>`-a exit,always -F arch=b32 -S unlink -S unlinkat -F key=file_deletion` |
+| Critical files changes   | `-w /etc/passwd -p wa -k passwd_changes`<br>`-w /etc/shadow -p wa -k shadow_changes`<br>`-w /etc/group -p wa -k group_changes`<br>`-w /etc/sudoers -p wa -k sudoers_changes`<br>`-w /etc/hosts -p wa -k hosts_changes` |
+| auth.log changes         | `-w /var/log/auth.log -p wa -k auth_logs` |
+| Network activity         | `-a exit,always -F arch=b64 -S connect,accept,sendto,recvfrom -F key=network_activity`<br>`-a exit,always -F arch=b32 -S connect,sendto,recvfrom -F key=network_activity` |
+| DNS changes              | `-w /etc/resolv.conf -p wa -k dns_changes` |
+| Socket activity          | `-a always,exit -F arch=b64 -S socket -F key=socket_activity`<br>`-a always,exit -F arch=b32 -S socket -F key=socket_activity` |
+| Time changes             | `-w /etc/localtime -p wa -k time_changes` |
+| Shutdown reboot          | `-w /var/log/wtmp -p wa -k shutdown_reboot` |
+| Cron changes             | `-w /etc/crontab -p wa -k cron_changes`<br>`-w /etc/cron.d -p wa -k cron_changes`<br>`-w /etc/cron.daily -p wa -k cron_changes`<br>`-w /etc/cron.hourly -p wa -k cron_changes`<br>`-w /etc/cron.weekly -p wa -k cron_changes`<br>`-w /etc/cron.monthly -p wa -k cron_changes` |
+| User group management    | `-a always,exit -F arch=b64 -S setuid,setresuid,setreuid,setfsuid,setgid,setresgid,setregid,setfsgid -F key=user_group_management`<br>`-a always,exit -F arch=b32 -S setuid,setresuid,setreuid,setfsuid,setgid,setresgid,setregid,setfsgid -F key=user_group_management` |
+| Modules                  | `-w /sbin/insmod -p x -k modules`<br>`-w /sbin/rmmod -p x -k modules`<br>`-w /sbin/modprobe -p x -k modules` |
+| Sudo usage               | `-w /var/log/sudo.log -p wa -k sudo_usage` |
+| Log changes              | `-w /var/log -p wa -k log_changes` |
+| Docker related           | `-w /usr/bin/dockerd -p wa -k docker_daemon`<br>`-w /etc/docker/daemon.json -p wa -k docker_config`<br>`-w /lib/systemd/system/docker.service -p wa -k docker_service`<br>`-w /lib/systemd/system/docker.socket -p wa -k docker_socket`<br>`-a always,exit -F arch=b64 -S execve -F path=/usr/bin/docker -k docker_commands`<br>`-w /var/lib/docker/ -p wa -k docker_storage`<br>`-a always,exit -F arch=b64 -S execve -F path=/usr/bin/docker -k network_activity`<br>`-a always,exit -F arch=b64 -S setuid,setgid,bind,connect -F comm="/usr/bin/docker" -k docker_sys` |
+
+### 3.3 CLI design
```

Comment:
>3.3 [](http://example.com/codeflow?start=4&length=3)

Do you have ConfigDB new schema design? If yes, you also need to define for the warm-reboot requirement. Please check [https://github.com/sonic-net/SONiC/blob/master/doc/hld_template.md](https://github.com/sonic-net/SONiC/blob/master/doc/hld_template.md) if any sections are missing.

Reply:
Updated